src/test/ui/async-await/async-fn-size.rs

```diff
@@ -89,10 +89,10 @@ fn main() {
     assert_eq!(8, std::mem::size_of_val(&await1_level1()));
     assert_eq!(12, std::mem::size_of_val(&await2_level1()));
     assert_eq!(12, std::mem::size_of_val(&await3_level1()));
-    assert_eq!(20, std::mem::size_of_val(&await3_level2()));
-    assert_eq!(28, std::mem::size_of_val(&await3_level3()));
-    assert_eq!(36, std::mem::size_of_val(&await3_level4()));
-    assert_eq!(44, std::mem::size_of_val(&await3_level5()));
+    assert_eq!(24, std::mem::size_of_val(&await3_level2()));
```

Comment:
cc @tmandry The generator transform is producing fewer `storage_conflicts` which is causing layout to pessimize from: `0: tag, 4: u16, 6: u16, 8: union { Future, Future, Future }`  to `0: tag, 4: u16, 8: union { Future, Future, (u16, Future) }`